go/api/request_routing_config.go

```diff
@@ -0,0 +1,189 @@
+	// Route request to all primary nodes.
+	SimpleNodeRouteAllPrimaries
+	// Route request to a random node.
+	// Warning: Don't use it with write commands, because they could be randomly routed to a replica (RO) node and fail.
+	SimpleNodeRouteRandom
+)
+
+func (simpleNodeRoute SimpleNodeRoute) toRoutesProtobuf() (*protobuf.Routes, error) {
+	simpleRouteProto, err := mapSimpleNodeRoute(simpleNodeRoute)
+	if err != nil {
+		return nil, err
+	}
+
+	request := &protobuf.Routes{
+		Value: &protobuf.Routes_SimpleRoutes{
+			SimpleRoutes: simpleRouteProto,
+		},
+	}
+	return request, nil
+}
+
+func mapSimpleNodeRoute(simpleNodeRoute SimpleNodeRoute) (protobuf.SimpleRoutes, error) {
+	switch simpleNodeRoute {
+	case SimpleNodeRouteAllNodes:
+		return protobuf.SimpleRoutes_AllNodes, nil
+	case SimpleNodeRouteAllPrimaries:
+		return protobuf.SimpleRoutes_AllPrimaries, nil
+	case SimpleNodeRouteRandom:
+		return protobuf.SimpleRoutes_Random, nil
+	default:
+		return protobuf.SimpleRoutes_Random, &RequestError{"Invalid simple node route"}
+	}
+}
+
+// Defines type of the node being addressed.
+type SlotType int
+
+const (
+	// Address a primary node.
+	SlotTypePrimary SlotType = iota
+	// Address a replica node.
+	SlotTypeReplica
+)
+
+func mapSlotType(slotType SlotType) (protobuf.SlotTypes, error) {
+	switch slotType {
+	case SlotTypePrimary:
+		return protobuf.SlotTypes_Primary, nil
+	case SlotTypeReplica:
+		return protobuf.SlotTypes_Replica, nil
+	default:
+		return protobuf.SlotTypes_Primary, &RequestError{"Invalid slot type"}
+	}
+}
+
+// Request routing configuration overrides the [api.ReadFrom] connection configuration.
+// If SlotTypeReplica is used, the request will be routed to a replica, even if the strategy is ReadFrom [api.PreferReplica].
+type SlotIdRoute struct {
+	// Defines type of the node being addressed.
+	slotType SlotType
+	// Slot number. There are 16384 slots in a Valkey cluster, and each shard manages a slot range.
+	// Unless the slot is known, it's better to route using [api.SlotTypePrimary].
+	slotID int32
+}
+
+func NewSlotIdRoute(slotType SlotType, slotId int32) *SlotIdRoute {
+	return &SlotIdRoute{slotType: slotType, slotID: slotId}
+}
+
+func (slotIdRoute *SlotIdRoute) toRoutesProtobuf() (*protobuf.Routes, error) {
+	slotType, err := mapSlotType(slotIdRoute.slotType)
+	if err != nil {
+		return nil, err
+	}
+
+	request := &protobuf.Routes{
+		Value: &protobuf.Routes_SlotIdRoute{
+			SlotIdRoute: &protobuf.SlotIdRoute{
+				SlotType: slotType,
+				SlotId:   slotIdRoute.slotID,
+			},
+		},
+	}
+	return request, nil
+}
+
+// Request routing configuration overrides the [api.ReadFrom] connection configuration.
+// If SlotTypeReplica is used, the request will be routed to a replica, even if the strategy is ReadFrom [api.PreferReplica].
+type SlotKeyRoute struct {
+	// Defines type of the node being addressed.
+	slotType SlotType
+	// The request will be sent to nodes managing this key.
+	slotKey string
```

Comment:
Are docs for private fields visible to a user? If no, consider copying docs to the ctor.